### PR TITLE
usb-modeswitch: 2.2.1 -> 2.5.0 and nixos module

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

We were carrying v2.2.1 and not v2.3.0 due to the update only touching the version and not the sha256 sum.

Additionally, we add a nixos module for setting up udev to use the rules.

Unfortunately, I no longer have the hardware that made me write the module, so I cannot actually test if it works. I thought it would be better to get things into nixpkgs instead of lingering around here...

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by peterhoeg** [comment]
@marcweber, any comments?

File: nixos/modules/hardware/usb_wwan.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  ###### interface
+
+  options = {
+
+    hardware.usb_wwan = {
```
Comment (fpletz) [inline nixos/modules/hardware/usb_wwan.nix]: The consensus is to use camelCase for nixos options instead of separators.

Comment (peterhoeg) [inline nixos/modules/hardware/usb_wwan.nix]: Changed. camelCase abbreviations just look really ugly though...

Comment (fpletz) [inline nixos/modules/hardware/usb_wwan.nix]: In this case, I agree. :)

**Review by fpletz — APPROVED** [review]

**Pull request merged**